apps/roboshield/src/payload/blocks/RobotsTxtGenerator.ts

```diff
@@ -1,7 +1,25 @@
-import { Block, Field } from "payload/types";
+import { Block, Field, Validate } from "payload/types";
 import richText from "../fields/richText";
 import { blocks } from "payload/dist/fields/validations";
 
+const validateSteps: Validate = (value = [], args) => {
+  const requiredSteps: string[] = ["finish"];
+  const missingSteps = requiredSteps.filter(
+    (slug) =>
+      !value?.find(
+        ({ blockType }: { blockType: string }) => blockType === slug,
+      ),
+  );
+  if (missingSteps.length) {
+    return `The following steps are missing: ${missingSteps.join(", ")}`;
+  }
+  const lastBlock = value[value.length - 1];
+  if (lastBlock?.blockType !== "finish") {
+    return "Finish Step should appear last";
+  }
```

Comment:
If there is existing-robots, we need to check that it's the first block... otherwise, a user can add it anywhere.

<img width="766" alt="Screenshot 2024-07-10 at 15 58 19" src="https://github.com/CodeForAfrica/ui/assets/1779590/ad68a374-6bb2-4f3f-aaa5-6bb09b6da00f">